core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -634,6 +637,20 @@ private SqlNode getIdentifier(SqlBasicCall intervalOperand) {
     return intervalOperand.operand(0);
   }
 
+  private void unparseOther(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    switch (call.getOperator().getName()) {
+    case "DAYOFYEAR":
+      final SqlWriter.Frame frame = writer.startFunCall("EXTRACT");
+      writer.sep("DAYOFYEAR FROM");
+      call.operand(0).unparse(writer, 0, 0);
```

Comment:
Instead of using the writer.sep make a SqlBasicCall to extract with appropriate Operand. There are multiple examples you can refer to